### PR TITLE
Fix BWC issue when reading Role.Properties from the cluster state

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
Relates to https://github.com/crate/crate/issues/16768

### CODE REVIEW

File: server/src/test/java/io/crate/role/RolePropertiesTest.java [hunk 1]
```diff
@@ -117,4 +124,20 @@ public void test_bwc_streaming() throws Exception {
...
+        xContentBuilder.endObject();
+
+        XContentParser parser = JsonXContent.JSON_XCONTENT.createParser(
+            NamedXContentRegistry.EMPTY,
+            DeprecationHandler.THROW_UNSUPPORTED_OPERATION,
+            Strings.toString(xContentBuilder));
+        Role.Properties properties = Role.Properties.fromXContent(parser);
+
+        BytesStreamOutput out = new BytesStreamOutput();
+        properties.writeTo(out);
```
Comment (BaurzhanSakhariev) [inline server/src/test/java/io/crate/role/RolePropertiesTest.java:141]: works with NULL, want a better test

File: docs/appendices/release-notes/5.9.2.rst [hunk 2]
```diff
@@ -47,4 +47,5 @@ See the :ref:`version_5.9.0` release notes for a full list of changes in the
 Fixes
 =====
 
-None
+- Fixed a BWC issue resulting in an exception when doing a rolling upgrade of a
+  cluster with custom users.
```
Comment (seut) [inline docs/appendices/release-notes/5.9.2.rst:51]:
```suggestion
- Fixed a BWC issue resulting in an exception when using custom user/roles created on a CrateDB version < :ref:`version_5.9.0`.
```

File: server/src/test/java/io/crate/role/RolePropertiesTest.java [hunk 3]
```diff
@@ -117,4 +124,21 @@ public void test_bwc_streaming() throws Exception {
...
+
+        XContentParser parser = JsonXContent.JSON_XCONTENT.createParser(
+            NamedXContentRegistry.EMPTY,
+            DeprecationHandler.THROW_UNSUPPORTED_OPERATION,
+            Strings.toString(xContentBuilder));
+        Role.Properties properties = Role.Properties.fromXContent(parser);
+
+        assertThat(properties.jwtProperties()).isNull();
+        assertThat(properties.password()).isNull();
+        assertThat(properties.sessionSettings()).isNotNull(); // Must be always not null as Sessions.newSession doesn't do a NULL check.
```
Comment (seut) [inline server/src/test/java/io/crate/role/RolePropertiesTest.java:142]: I don't think we need to comment on basic constraints and usage expectations.
Also, why not expect what we want?
```suggestion
        assertThat(properties.sessionSettings()).isEqualTo(Map.of()); 
```

**Comment by henrikingo** [comment]
Let's not reference a private ticket from a public commit. Please either reference 16768 or a new crate/crate issue.

File: docs/appendices/release-notes/5.9.2.rst (hunk 2 above)
Comment (BaurzhanSakhariev) [reply docs/appendices/release-notes/5.9.2.rst]: Thanks for the suggestion. Applied and adjusted to fit in 80 symbols per line.

**Review by seut — APPROVED** [review]
thx

**Pull request merged**